MeetupPlatformApi/Controllers/AuthenticationController.cs

```diff
@@ -44,33 +44,99 @@ public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto reg
         }
 
         var user = mapper.Map<UserEntity>(registrationDto);
-
         user.Password = BCrypt.HashPassword(user.Password);
+        await context.Users.AddAsync(user);
 
-        context.Users.Add(user);
+        var tokenPair = authenticationManager.IssueTokenPair(user);
+        var refreshToken = new RefreshTokenEntity()
+        {
+            Id = authenticationManager.GetNameIdentifier(tokenPair.RefreshToken),
+            Expires = authenticationManager.GetExpires(tokenPair.RefreshToken),
+            UserId = user.Id
+        };
+        await context.RefreshTokens.AddAsync(refreshToken);
         await context.SaveChangesAsync();
 
-        var accessToken = authenticationManager.IssueAccessToken(user);
         var userInfoDto = mapper.Map<UserOutputDto>(user);
         var outputDto = new UserRegistrationResultDto
         {
             UserInfo = userInfoDto,
-            AccessToken = accessToken
+            TokenPair = new TokenPairDto
+            {
+                AccessToken = tokenPair.AccessToken,
+                RefreshToken = tokenPair.RefreshToken
+            }
         };
         return CreatedAtAction(nameof(GetUserById), new {id = outputDto.UserInfo.Id}, outputDto);
     }
 
+    [HttpPost("refresh-tokens")]
+    public async Task<IActionResult> RefreshTokenPair([FromBody] RefreshTokenDto refreshToken)
+    {
+        var refreshTokenId = authenticationManager.GetNameIdentifier(refreshToken.RefreshToken);
+        var refreshTokenInfo = await context.RefreshTokens.SingleOrDefaultAsync(refreshToken => refreshToken.Id == refreshTokenId);
+
+        if (refreshTokenInfo is null)
+        {
+            return NotFound($"Token with id: {refreshTokenId} doesn't exist.");
```

Comment:
Consider either removing this message or replacing it with more generalized/descriptive one, for example `"Provided token is either fake or already was used"`.